test/utils/test-streams.js

```diff
@@ -66,10 +68,62 @@ function slowCount(value) {
   });
 }
 
+/*
+ * Creates a transform stream which joins objects from an object stream into an array.
+ * At end-of-stream, optional callback is invoked with array of collected objects.
+ */
+function join(cb) {
```

Comment:
I'm not thrilled with this name, but since it has different semantics from the previously used `miss.concat()` it seemed like a good idea to change the name.